collect_app/src/test/java/org/odk/collect/android/utilities/MultiFormDownloaderTest.java

```diff
@@ -1,414 +0,0 @@
-     */
-    @Test
-    public void downloadingFormWithMedia_Succeeds() throws Exception {
-        String basicMedia = "<h:html xmlns=\"http://www.w3.org/2002/xforms\" xmlns:h=\"http://www.w3.org/1999/xhtml\">\n" +
-                "    <h:head>\n" +
-                "        <h:title>basic-media</h:title>\n" +
-                "        <model>\n" +
-                "            <instance>\n" +
-                "                <data id=\"basic-media\">\n" +
-                "                    <q1/>\n" +
-                "                </data>\n" +
-                "            </instance>\n" +
-                "            <bind nodeset=\"/data/q1\" type=\"string\"/>\n" +
-                "            <itext> \n" +
-                "                <translation default=\"true()\" lang=\"English\">\n" +
-                "                    <text id=\"/data/q1:label\"><value form=\"image\">jr://images/b.jpg</value></text>\n" +
-                "                </translation>\n" +
-                "            </itext>\n" +
-                "        </model>\n" +
-                "    </h:head>\n" +
-                "    <h:body>\n" +
-                "        <input ref=\"/data/q1\">\n" +
-                "            <label>Question</label>\n" +
-                "        </input>\n" +
-                "    </h:body>\n" +
-                "</h:html>";
-        File formXml = File.createTempFile("basicMedia", ".xml");
-        formXml.deleteOnExit();
-
-        BufferedWriter out = new BufferedWriter(new FileWriter(formXml));
-        out.write(basicMedia);
-        out.close();
-
-        MultiFormDownloader downloader = spy(new MultiFormDownloader(openRosaXMLFetcher));
-        ServerFormDetails serverFormDetails = new ServerFormDetails("Media", "https://testserver/media.xml",
-                "https://testserver/media-manifest.xml", "media", "2019121201",
-                "hash", "manifestHash", false, false);
-        MultiFormDownloader.FileResult result = new MultiFormDownloader.FileResult(formXml, true);
-        doReturn(result).when(downloader).downloadXform(serverFormDetails.getFormName(), serverFormDetails.getDownloadUrl(), null);
-        doReturn("").when(downloader).downloadManifestAndMediaFiles(any(), any(), any(), anyInt(), anyInt(), any());
-        doReturn(true).when(downloader).installEverything(any(), any(), any());
-
-        List<ServerFormDetails> forms = new ArrayList<>();
-        forms.add(serverFormDetails);
-
-        HashMap<ServerFormDetails, String> messages = downloader.downloadForms(forms, null);
-        assertThat(messages.get(serverFormDetails), is("Success"));
-    }
-
-    /**
-     * Forms with references to external secondary instance need to have the secondary instance
-     * available at time of form parse.
-     *
-     * See https://github.com/getodk/collect/issues/3635
-     */
-    @Test
-    public void downloadingFormWithXmlExternalSecondaryInstance_Succeeds() throws Exception {
```

Comment:
I think this case should get coverage somewhere. I think the CSV case can be skipped but the last-saved one should also be captured.

Reply:
Same as the media file progress test I think. Didn't need it but I think you're right we should add it in any way.

Reply:
Sorry I mean "same story", not "same code.